backends/tfhe-cuda-backend/cuda/include/pbs/pbs_utilities.h

```diff
@@ -86,7 +86,7 @@ template <typename Torus> struct pbs_buffer<Torus, PBS_TYPE::CLASSICAL> {
     cuda_set_device(gpu_index);
     this->pbs_variant = pbs_variant;
 
-    auto max_shared_memory = cuda_get_max_shared_memory(gpu_index);
+    uint max_shared_memory = (uint)(cuda_get_max_shared_memory(gpu_index));
```

Comment:
This could be a good opportunity to actually change the return type of `cuda_get_max_shared_memory`. It should be `unsigned`, not `int`.

Reply:
Yes totally. Then I think I will use a type u32 for max_shared_memory, as we have to pass it across the FFI.